src/Inventory/Asset/Software.php

```diff
@@ -818,13 +842,38 @@ public function logSoftwares()
         }
 
         foreach ($this->deleted_versions as $software_data) {
+            $softwares_id  = $software_data['softid']    ?? "";
+            $versions_id   = $software_data['versionid'] ?? "";
+            $software_name = $software_data['name']      ?? "";
+            $version_name  = $software_data['version']   ?? "";
+            $asset_name    = $this->item->fields['name'] ?? "";
```

Comment:
All these variables are defined at this point.
```suggestion
            $softwares_id  = $software_data['softid'];
            $versions_id   = $software_data['versionid'];
            $software_name = $software_data['name'];
            $version_name  = $software_data['version'];
            $asset_name    = $this->item->fields['name'];
```